maxMoviesInYear.js

```diff
@@ -0,0 +1,50 @@
+const mockData= require('./mockResult.js')
+
+function findMaxMoviesInYear () {
+    const moviesInAYear = {};
+    for ( let movie of mockData ) {
+        movie.license.reduce( ( acc, currentValue,currentIndex ) => {  
+            let currentYear = currentValue;
+
+            // loop through the license array till the end of the array and not consider the second index of the license array since it would be duplicate entry.
+
+            while ( currentYear <= movie.license[movie.license.length - 1] && (currentIndex!==movie.license.length-1) ) {  
+                if ( moviesInAYear[currentYear] ) {
+                    moviesInAYear[currentYear] += 1;
+                } else {
+                    moviesInAYear[currentYear] = 1;
+                }
+                currentYear += 1;
+            }
+            return moviesInAYear
+        },moviesInAYear)
+    }
+    return findMaxKeyInAObject(moviesInAYear)
+  
+}
+
+
+function findMaxKeyInAObject ( obj ) {
```

Comment:
Needs a better name

Reply:
Done!! Changed to 'getKeysWithMaxValuesInObject'